frontend/src/app/core/state/hal-resource.ts

```diff
@@ -0,0 +1,19 @@
+export interface HalResourceLink {
```

Comment:
All this stuff should perhaps be in this file https://github.com/opf/openproject/pull/9581/files#diff-c6bb53f5fd6455e2e634599268c6fbc03f05b27d5db284e36528c597cffde956R1

But I'm not sure yet.